src/main/java/com/zendesk/maxwell/MaxwellConfig.java

```diff
@@ -517,6 +520,15 @@ private void setup(OptionSet options, Properties properties) {
 		if (outputConfig.encryptionEnabled()) {
 			outputConfig.secretKey = fetchOption("secret_key", options, properties, null);
 		}
+
+		if (this.kafkaPartitionConfigFile != null) {
+			kafkaPartitionMap = new HashMap<>();
+			Properties partitionProps = readPropertiesFile(this.kafkaPartitionConfigFile, true);
+
+			for (String databaseName: partitionProps.stringPropertyNames()) {
+				kafkaPartitionMap.put(databaseName, Integer.parseInt(partitionProps.getProperty(databaseName)));
+			}
+		}
```

Comment:
this code should be moved to `validate` -- `setup` is not called in embedded mode.